src/Expander.hs

```diff
@@ -164,41 +166,65 @@ getImports (PrefixImports spec pref) =
 
 visit :: ModuleName -> Expand Exports
 visit modName = do
-  (world', m, es) <-
+  (m, es) <-
     do world <- view expanderWorld <$> getState
        case view (worldModules . at modName) world of
          Just m -> do
            let es = maybe noExports id (view (worldExports . at modName) world)
-           return (world, m, es)
+           return (m, es)
          Nothing -> do
            (m, es) <- inPhase runtime $ loadModuleFile modName
-           w <- view expanderWorld <$> getState
-           return (w, m, es)
+           return (m, es)
   p <- currentPhase
   let i = phaseNum p
   visitedp <- Set.member p .
               view (expanderWorld . worldVisited . at modName . non Set.empty) <$>
               getState
   unless visitedp $ do
-    let m' = shift i m
-    let envs = view worldEnvironments world'
-    let tenvs = view worldTransformerEnvironments world'
-    (moreEnvs, transformerEnvs, evalResults) <- expandEval $ evalMod envs tenvs p m'
-    modifyState $
-      set (expanderWorld . worldVisited . at modName . non Set.empty . at p)
-          (Just ())
-    modifyState $
-      over (expanderWorld . worldEnvironments) (Map.unionWith (<>) moreEnvs)
-    modifyState $
-      over (expanderWorld . worldTransformerEnvironments) (Map.unionWith (<>) transformerEnvs)
-
+    let m' = shift i m -- Shift the syntax literals in the module source code
+    evalResults <- inPhase p $ evalMod m'
     modifyState $
       set (expanderWorld . worldEvaluated . at modName)
           (Just evalResults)
-
-
   return (shift i es)
 
+-- | Evaluate an expanded module at the current expansion phase,
```

Comment:
What caused this move? Are there any changes in the code, or did it just migrate?

Reply:
It had to move because the handing of imports changed. Before, evaluating a module with imports in it just ignored the declared imports. But what actually needs to happen is to load those imports with respect to the phase at which we're evaluating the module. Loading the imports requires visiting the module, and visiting is in the expander monad rather than the evaluator monad. Hence the new location.

The resulting code is also simpler in the expander monad. A lot of the tracking of phases and environments with the `RWST` transformer on top of the expander was really simulating things that the expander does anyway.

Reply:
Here's the most important line of that change: https://github.com/gelisam/klister/pull/24/files/883e8954624a0c2c84cdbc129dcd4a2335ed516f#diff-a63cde2fb12e931100bbb12fc30e04eeR226